main/drivers/CANDriver.c

```diff
@@ -0,0 +1,293 @@
+    driver.messageEvents = xEventGroupCreate();
+    driver.globalEvents = globalEvents;
+    driver.messageFlag = messageFlag;
+
+    // Initialize device structs
+    CANDevice_t *dev;
+    for (int i = 0; i < CAN_BUS_COUNT; i++) {
+        dev = &driver.devices[i];
+        dev->statsMutex = xSemaphoreCreateMutex();
+    }
+
+    // Create driver task on core 1
+    xTaskCreatePinnedToCore(CANDriverTask, "CANDriver", CAN_STACK_SIZE, NULL, DEFAULT_TASK_PRIORITY,
+                            &driver.driverTaskHandler, 1);
+}
+
+/**
+ * Add a new CAN bus chip to the driver
+ * @param bus the bus number, must be unique to each chip and at max CAN_BUS_COUNT
+ * @param csPin The pin nCS of the MCP251863
+ * @param interruptPin The nINT1 pin of the MCP251863
+ * @param standbyPin The STBY pin of the MCP251863 (optional, set -1 to ignore)
+ */
+int canDriverAddBus(uint8_t bus, int csPin, int interruptPin, int standbyPin) {
+    if (bus >= CAN_BUS_COUNT)
+        return -1;
+
+    // setup device info
+    CANDevice_t *dev = &driver.devices[bus];
+    dev->CSPin = csPin;
+    dev->InteruptPin = interruptPin;
+    memset(&dev->stats, 0, sizeof(CANDeviceStatistic_t));
+
+    MCP251XFD mcp2517fd = DEFAULT_MCP251863_DRIVER_CONFIG(&dev->spi, csPin);
+    mcp2517fd.DriverConfig |= MCP251XFD_DRIVER_INIT_SET_RAM_AT_0;
+    memcpy(&dev->mcp251863, &mcp2517fd, sizeof(MCP251XFD));
+
+    // configure MCP251XFD
+    MCP251XFD_Config MCP251863_CAN_conf =
+        DEFAULT_MCP251863_CONTROLLER_CONFIG(&dev->MCP251863_SYSCLK1, &dev->MCP251863_BTStats);
+    MCP251863_CAN_conf.SysInterruptFlags =
+        (MCP251XFD_INT_RX_EVENT | MCP251XFD_INT_RX_OVERFLOW_EVENT | MCP251XFD_INT_CLEARABLE_FLAGS_MASK);
+
+    // Setup FIFO1 for receiving and FIFO2 fro transmitting
+    MCP251XFD_FIFO MCP251863_CAN_FIFO_Conf[CAN_FIFO_COUNT] = {
+        {
+            .Name = MCP251XFD_FIFO1,
+            .Size = MCP251XFD_FIFO_32_MESSAGE_DEEP,
+            .Payload = CAN_RX_PAYLOAD,
+            .Direction = MCP251XFD_RECEIVE_FIFO,
+            .ControlFlags = MCP251XFD_FIFO_NO_TIMESTAMP_ON_RX,
+            .InterruptFlags = MCP251XFD_FIFO_OVERFLOW_INT | MCP251XFD_FIFO_RECEIVE_FIFO_HALF_FULL_INT |
+                              MCP251XFD_FIFO_RECEIVE_FIFO_NOT_EMPTY_INT,
+            .RAMInfos = NULL,
+        }, // SID: 0x000..0x1FF ; No EID
+        {
+            .Name = MCP251XFD_FIFO2,
+            .Size = MCP251XFD_FIFO_4_MESSAGE_DEEP,
+            .Payload = MCP251XFD_PAYLOAD_8BYTE,
+            .Direction = MCP251XFD_TRANSMIT_FIFO,
+            .Attempts = MCP251XFD_THREE_ATTEMPTS,
+            .Priority = MCP251XFD_MESSAGE_TX_PRIORITY16,
+            .ControlFlags = MCP251XFD_FIFO_NO_RTR_RESPONSE,
+            .InterruptFlags = MCP251XFD_FIFO_TX_ATTEMPTS_EXHAUSTED_INT | MCP251XFD_FIFO_TRANSMIT_FIFO_NOT_FULL_INT,
+            .RAMInfos = NULL,
+        },
+    };
+
+    // configure the interupt pin
+    gpio_config_t pinConf = {};
```

Comment:
I prefer this setup, but it doesn't really matter
```suggestion
    gpio_config_t pinConf = {
        .mode = GPIO_MODE_INPUT;
        ...
    };
```